.github/workflows/build.yml

```diff
@@ -184,16 +184,17 @@ jobs:
           java-version: 17
           distribution: temurin
           cache: 'gradle'
-      - name: Gradle toolchains
-        uses: gradle/gradle-build-action@v2
+      - name: Set up Gradle
+        uses: gradle/actions/setup-gradle@v3
         with:
           gradle-version: release-candidate
-          arguments: -Porg.gradle.java.installations.auto-download=false javaToolchains
+
+      - name: Gradle version
+        run: gradle --version
+      - name: Gradle toolchains
+        run: gradle -Porg.gradle.java.installations.auto-download=false javaToolchains
       - name: Gradle build
-        uses: gradle/gradle-build-action@v2
-        with:
-          gradle-version: release-candidate
-          arguments: --refresh-dependencies -PexperimentalJavaVersion=${{ env.EXPERIMENTAL_JAVA }} -PjunitVersion=${{ matrix.junit-version }} -Porg.gradle.java.installations.auto-download=false --stacktrace --scan clean build -x spotlessCheck
+        run: gradle --refresh-dependencies -PexperimentalJavaVersion=${{ env.EXPERIMENTAL_JAVA }} -PjunitVersion=${{ matrix.junit-version }} -Porg.gradle.java.installations.auto-download=false --stacktrace --scan clean build -x spotlessCheck
```

Comment:
note that these two are not running `./gradlew` but `gradle` from the PATH, because [`with:gradle-version:` sets that up](https://github.com/gradle/actions/blob/v3/setup-gradle/action.yml#L7-L9)

Reply:
is there a way to run this on a PR?

Reply:
Thanks for adding full-build.